### PR TITLE
Solved Day 02 in C++

### PR DESCRIPTION
**Pull request by Celebrian**

### CODE REVIEW

**Comment by Celebrian** [comment]
Unsure if you want me in the future to add the running of tests like you did after I added the day one solution here: 174723e861 
It have not been mentioned in CONTRIBUTING.md so I have not done it so far. 

I really appreciate these the effort behind this every year, and it's a nice opportunity to brush off my rusty programming skills :D We will see how many days I can keep up :D

File: days/day-02/solutions/day02.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <cstring>
+#include <sstream>
+
+int main() {
+  const int MAXLINELENGTH = 30;
+  //Number of valid passwords
+  int valid_passwords = 0;
+  int acctually_valid_passwords = 0;
```
Comment (Arxcis) [inline days/day-02/solutions/day02.cpp:9]: Haha! Really really valid passwords? :+1: Good work on this one, thank you :smiley:

**Review by Arxcis — APPROVED** [review]

**Comment by Arxcis** [comment]
> Unsure if you want me in the future to add the running of tests like you did after I added the day one solution here: [174723e](https://github.com/Arxcis/adventofcode2020/commit/174723e8618ec6a5b07ded5c6998b58637d07087)

Yes I want :pray: 
I will update the CONTRIBUTING.md :+1: I must have forgot to add it.

Because then we can test your solution in the CI, before we merge :rocket:

**Comment by Arxcis** [comment]
Feel free to add the test now, if you got time, and know what to do :+1:

File: days/day-02/solutions/day02.cpp (hunk 1 above)
Comment (Celebrian) [reply days/day-02/solutions/day02.cpp:9]: Hehe, that is what happens when I want to solve part one first, then the second part "replaces" the correct answer in part one 😄 I'm honestly not that happy with how I read the input with stringline on line 18-20, but I failed when I tried to use sscanf like you did in your C solution. I'm expecting my solutions to become more and more janky 😄

Comment (Arxcis) [reply days/day-02/solutions/day02.cpp:9]: No worries :laughing: In the end, what matters is what is delivered to `stdout` :+1: The rest is just details :wink:

File: days/day-02/test.sh [hunk 2]
```diff
@@ -6,4 +6,5 @@ D=$(dirname $(realpath $0))
 echo ""
 echo "--- Day 2: Password Philosophy ---"
 $D/../../languages/c.sh $D/input.txt $D/output.txt $D/solutions/day02.c
+$D/../../languages/c.sh $D/input.txt $D/output.txt $D/solutions/day02.cpp
```
Comment (Arxcis) [inline days/day-02/test.sh]: Wups! It should be `cpp.sh` here :eyes: 
```suggestion
$D/../../languages/cpp.sh $D/input.txt $D/output.txt $D/solutions/day02.cpp
```

**Review by Arxcis — CHANGES_REQUESTED** [review]

File: days/day-02/test.sh (hunk 2 above)
Comment (Celebrian) [reply days/day-02/test.sh]: Ah, my bad:D, will fix:D That's what happens when I just copy/paste:D

**Review by Arxcis — CHANGES_REQUESTED** [review]
Sorry @Celebrian but I created a conflict on your branch, by meging another branch first. It should be an easy fix :+1:

**Comment by Arxcis** [comment]
Green Green Green ! merging...

**Pull request merged**